sdks/go/pkg/beam/metrics.go

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package beam
+
+import (
+	"context"
+
+	"github.com/apache/beam/sdks/go/pkg/beam/core/metrics"
+)
+
+// Counter is a metric that can be incremented and decremented,
+// and is aggregated by the sum.
+type Counter struct {
```

Comment:
Would

   type Couter = metrics.Counter

work?

Reply:
It would work, but it would make the go doc less complete.
The definition would be displayed (the type alias), but methods on the aliased types are not forwarded to the aliasing godoc. Instead, a link to the original document definition would be provided.

Further, since the methods don't exist in the godoc, examples couldn't be "hung" off of the methods, they'd need to be a part of the type only.

We could work around this by having the method described in formatted code snippets, but if the API changes, there's a risk the documentation won't be updated, and it won't break.

Reply:
Makes sense. What you have LGTM.